### PR TITLE
Feature/ui api private group

### LINKED ISSUE #1622: UI for making API private for a user group
In addition to being able to select between private and public, owner should be able to select that the API is public/visible to certain users: "private to a group".

# Definition of done
* [x] In the API profile, the owner can select that the API is private to a group.
* [ ] In the API Catalog, the owner can see an indication of 
  * [ ] completely private API
  * [ ] API being partially provate (visible to a selected group of users).

# Wireframes

1. API Visibility comes under Settings tab.
Has a separate Pane with radio buttons to select visibility option.
Default is public.
closedapi

2. To give access to specific people.
The section gets active when radio button is selected
Users can be added via email.
On adding a new user, they should appear as tags under the text field.
In API profile page , and API catalog, special icon should appear to indicate the API is visible for a specific group
(reference of BS tags: https://bootstrap-tagsinput.github.io/bootstrap-tagsinput/examples/)
closedapi2

3. API is accessible only to owner
In API profile page , and API catalog, special icon should appear to indicate the API is visible only for owner
closedapi3

### PR DESCRIPTION
**Pull request by nnn**
Closes #1622 

@brylie you have a task to make this UI work. I propose not to merge this PR, just to review. You can continue to work with it after review.

### CODE REVIEW

**Comment by nnn** [comment]
After discussion with @brylie and @bajiat will continue on with this task.

TODO:
- [ ] Make all three options functional; user is able to choose between private, limited private and public
- [ ] Indicate the visibility status of the API (ex. with a label next the API name)

**Comment by brylie** [comment]
Here are some examples of using camelCase allowedValues, and adding corresponding i18n strings:

- [apiUmbrellaSchema.rate_limit](https://github.com/apinf/platform/blob/1fa2668de38d0de343c64ae14c50e7e39ce3db53/proxy_backends/collection/apiUmbrellaSchema.js#L41-L43)
  - [i18n strings](https://github.com/apinf/platform/blob/develop/core/lib/i18n/en.i18n.json#L407-L409)

**Comment by jykae** [comment]
@nnn Are you doing UI indicator as part of this task? Just need to know we are not duplicating efforts #1624

**Comment by nnn** [comment]
@jykae no, I was not planning. Does your #1624 is blocked by my PR?

**Comment by brylie** [comment]
@bajiat do we need an extra task for adding UI indicator(s) for API privacy status?

**Comment by jykae** [comment]
@brylie I can do it on my task, would not want this to be put awfully small pieces.

File: apis/client/view/visibility/visibility.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template name="visibilityToggle">
-  {{#autoForm collection=formCollection doc=api id="apiVisibility" type=formType }}
+  {{#autoForm collection=ApisCollection doc=api id="apiVisibility" type="update" }}
     {{> afQuickField name='visibility' options="allowed" }}
```
Comment (jykae) [inline apis/client/view/visibility/visibility.html]: @nnn you can use firstOption=false to remove "Select one" option

**Comment by bajiat** [comment]
All the commits have been included in #1697. Closing this PR.

**Pull request closed without merging**